### PR TITLE
Release v0.11.0

### PR DESCRIPTION
**Pull request by AurevoirXavier**
https://github.com/darwinia-network/substrate-update-tracking/issues/19

### CODE REVIEW

File: runtime/crab/src/lib.rs [hunk 1]
```diff
@@ -140,7 +140,8 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: sp_runtime::create_runtime_str!("Crab"),
 	impl_name: sp_runtime::create_runtime_str!("Darwinia Crab"),
 	authoring_version: 0,
-	spec_version: 43,
+	// crate version ~0.11.0 := >=0.11.0, <0.12.0
+	spec_version: 1100,
```
Comment (AsceticBear) [inline runtime/crab/src/lib.rs:144]: Why not 44？

File: runtime/darwinia/src/lib.rs [hunk 2]
```diff
@@ -187,7 +187,8 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: sp_runtime::create_runtime_str!("Darwinia"),
 	impl_name: sp_runtime::create_runtime_str!("Darwinia"),
 	authoring_version: 0,
-	spec_version: 24,
+	// crate version ~0.11.0 := >=0.11.0, <0.12.0
+	spec_version: 1100,
```
Comment (AsceticBear) [inline runtime/darwinia/src/lib.rs:192]: Why not 25?

File: runtime/crab/src/lib.rs (hunk 1 above)
Comment (AurevoirXavier) [reply runtime/crab/src/lib.rs:144]: New style. More readable.

File: node/service/src/chain_spec/crab.rs [hunk 3]
```diff
@@ -135,7 +135,7 @@ pub fn crab_build_spec_genesis() -> GenesisConfig {
 			epoch_config: Some(BABE_GENESIS_EPOCH_CONFIG),
 		},
 		pallet_indices: Default::default(),
-		darwinia_balances_Instance0: BalancesConfig {
+		darwinia_balances_Instance2: BalancesConfig {
```
[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply node/service/src/chain_spec/crab.rs]: Seems you're reviewing on a wrong commit.

Refresh to pull the latest code.

[the remark this answers is not in the document]
Comment (AsceticBear) [reply node/service/src/chain_spec/crab.rs]: Hmm, I review trace the commits in sequence, and write this comment in ahead commit, you fixed this problem later.

**Comment by HackFisher** [comment]
Let's change the mmr sign/confirm threshold to 3/5 (60%).

File: runtime/crab/src/pallets/staking.rs [hunk 4]
```diff
@@ -2,31 +2,41 @@
...
+}
+#[cfg(not(feature = "dev"))]
 frame_support::parameter_types! {
-	pub const StakingModuleId: ModuleId = ModuleId(*b"da/staki");
-	pub const SessionsPerEra: SessionIndex = SESSIONS_PER_ERA;
 	pub const BondingDurationInEra: EraIndex = 14 * DAYS
 		/ (SESSIONS_PER_ERA as BlockNumber * BLOCKS_PER_SESSION);
 	pub const BondingDurationInBlockNumber: BlockNumber = 14 * DAYS;
 	// slightly less than 14 days.
+	#[cfg(not(feature = "dev"))]
```
Comment (HackFisher) [inline runtime/crab/src/pallets/staking.rs:24]: Why we need this line?

File: runtime/darwinia/src/pallets/election_provider_multi_phase.rs [hunk 5]
```diff
@@ -1,13 +1,17 @@
...
 
 sp_npos_elections::generate_solution_type!(
 	#[compact]
-	pub struct NposCompactSolution16::<u32, u16, sp_runtime::PerU16>(16)
-	// -------------------- ^^ <NominatorIndex, ValidatorIndex, Accuracy>
+	pub struct NposCompactSolution16::<
+		VoterIndex = u32,
+		TargetIndex = u16,
+		Accuracy = PerU16,
+	>(16)
```
Comment (HackFisher) [inline runtime/darwinia/src/pallets/election_provider_multi_phase.rs:14]: Why crab is using ```NposCompactSolution24```, but darwinia is using ```NposCompactSolution16```

File: runtime/darwinia/src/pallets/staking.rs [hunk 6]
```diff
@@ -2,31 +2,42 @@
...
+}
+#[cfg(not(feature = "dev"))]
 frame_support::parameter_types! {
-	pub const StakingModuleId: ModuleId = ModuleId(*b"da/staki");
-	pub const SessionsPerEra: SessionIndex = SESSIONS_PER_ERA;
 	pub const BondingDurationInEra: EraIndex = 14 * DAYS
 		/ (SESSIONS_PER_ERA as BlockNumber * BLOCKS_PER_SESSION);
 	pub const BondingDurationInBlockNumber: BlockNumber = 14 * DAYS;
 	// slightly less than 14 days.
+	#[cfg(not(feature = "dev"))]
```
Comment (HackFisher) [inline runtime/darwinia/src/pallets/staking.rs:24]: why we need this?

**Review by HackFisher — CHANGES_REQUESTED** [review]
@AsceticBear Please help review this.

Need confirmation for node testing and try_runtime from @AurevoirXavier @AsceticBear 

Need companion PR for apps and darwinia.js from @sxlwar @WoeOm

File: runtime/darwinia/src/pallets/election_provider_multi_phase.rs (hunk 5 above)
Comment (AurevoirXavier) [reply runtime/darwinia/src/pallets/election_provider_multi_phase.rs:14]: Kusama: `NposCompactSolution24`
Polkadot: `NposCompactSolution16`

And I suggest increase Crab's validator count.

**Review by HackFisher — APPROVED** [review]

**Comment by AsceticBear** [comment]
Something worth note, if somebody upgrading the runtime directly without update the node binary will result in the following error:

```sh
2021-06-09 19:08:42 Unable to author block in slot 270539487,. `can_author_with` returned: Failed to get runtime version at `BlockId::Hash(0xcdc6ce80c7729001661c521e429ab83f6c060dfd4307cf3eaa0e4ee3eef2be96)` and will disable authoring. Error: VersionInvalid("RuntimeConstruction(Instantiation(\"Instantiation: Export ext_logging_max_level_version_1 not found\"))") Probably a node update is required!    
```

**Review by AsceticBear — APPROVED** [review]

**Comment by HackFisher** [comment]
@WoeOm @sxlwar I merged this now, but please provide the link of the companion PRs which was expected to included before this PR merge.

@AurevoirXavier Please check the try_runtime tests.

For #665 , let's do it in next runtime upgrade after further confirmation.

**Comment by HackFisher** [comment]
And, @AurevoirXavier please bump crates version and update CHANGELOG.md before do github release.

**Pull request merged**